docs/administration/managing-infrastructure/rotate-master-key.md

```diff
@@ -0,0 +1,77 @@
+
+## What is affected by the rotation
+
+Octopus [encrypts important and sensitive data](/docs/security/data-encryption.md) using a Master Key. This includes:
+
+- The Octopus Server X.509 certificate which is used for [Octopus to Tentacle communication](/docs/security/octopus-tentacle-communication/index.md) - your Tentacles should still trust your Octopus Server after the rotation.
+- Sensitive variable values, wherever you have defined them.
+- Sensitive values in your deployment processes, like the password for a custom IIS App Pool user account.
+- Sensitive values in your deployment targets, like the password for creating [Offline Drops](/docs/infrastructure/deployment-targets/offline-package-drop.md).
+
+## Rotating the Master Key
+
+### Step 1. Back up before you start
+
+Make sure to [back up everything](/docs/administration/data/backup-and-restore.md) before you start this process. This should also include the `OctopusServer.config` which contains the old master key in case the process fails.
+
+### Step 2. Stop the server
+
+Master key rotation currently only works when Octopus server is stopped, as the database will not be in a valid state during the rotation process. In an HA cluster, this means all the server nodes need to be stopped. You can do this with `Octopus.Server service --stop` on each server node. This also ensures anything in memory or in transit is persisted to disk before we start.
+
+### Step 3. Rotate the master key
+
+Once everything is backed up and the Octopus Server stopped, the steps are as follows.
+
+1. Run `Octopus.Server rotate-master-key` and follow the prompts. This will guide you through the steps and generate a report at the end of the process. A new master key will also be written to its own file. In an HA setup, this command can be run from any server node.
+1. If you have an HA setup, run `Octopus.Server set-master-key --masterKey=NEW_MASTER_KEY` on the other server nodes.
+1. You can confirm the new master key is being used by running `Octopus.Server show-master-key`.
+1. Run `Octopus.Server service --start` to start the Octopus Server running against the rotated database.
+
+:::warning
+**Please read the report carefully and get in touch with us if anything seems out of the ordinary. Back up your new Master Key!**
+:::
+
+Here's the beginning of an example report:
+
+```text
+================================================================================
+ROTATE MASTER KEY REPORT
+================================================================================
+New Master Key: Gj3GfVf1gLn8kQA7wX4iXw==
+Processed 10533 documents.
+10009/10533 documents were updated
+0/10533 documents were left unchanged due to errors
```

Comment:
@zentron @veochen-octopus we are looking at using this in cloud instances, and I am unsure what this line in the report means? Does this mean some documents won't decrypt, do we need to rollback?

Reply:
I admit this line reads a bit weird. It's like this for now to be consistent with the `lost-master-key` report. It just means there are 0 errors. In reality, if we do get an error, it _should_ automatically rollback the entire transaction. This has been tested. So we should never see this number go above 0. If that happens then we have a problem.